Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLink.cs

```diff
@@ -545,10 +561,21 @@ private void Update()
 
                 // Used to correct for the volume of the audio source component
 
-                audioMaterial.SetFloat(_SourceVolume, audioSource.volume);
-                audioMaterial.SetFloat(_SourceSpatialBlend, audioSource.spatialBlend);
-                audioMaterial.SetVector(_SourcePosition, audioSource.transform.position);
+                float sourceVolume = audioSource.volume;
+                Vector3 sourcePosition = audioSource.transform.position;
+                float sourceMax = audioSource.maxDistance;
 
+                AnimationCurve sourceBlendCurve = audioSource.GetCustomCurve(AudioSourceCurveType.SpatialBlend);
+                AnimationCurve sourceFalloffCurve = audioSource.GetCustomCurve(AudioSourceCurveType.CustomRolloff);
+                var playerEars = GetHearingLocation();
+                float listeningDistance = Vector3.Distance(playerEars, sourcePosition) / sourceMax;
+                float sourceFalloff = sourceFalloffCurve.Evaluate(listeningDistance);
```

Comment:
Does this just clamp when the input is greater than 1? Can happen when past the max distance, right?